samcli/cli/types.py

```diff
@@ -204,14 +204,19 @@ def convert(self, value, param, ctx):
         value = (value,) if not isinstance(value, tuple) else value
 
         for val in value:
-            groups = re.findall(self._pattern, val)
-
-            if not groups:
-                fail = True
-            for group in groups:
-                key, v = group
-                # assign to result['KeyName1'] = string and so on.
-                result[_unquote_wrapped_quotes(key)] = _unquote_wrapped_quotes(v)
+            # Using standard parser first. We should implement other type parser like JSON and Key=key,Value=val type format.
+            parsed, k, v = self._standard_key_value_parser(val)
```

Comment:
should we check for more than 1 count, given that aws cli already does it.

Reply:
aws-cli simply pick first `=`. I think our existing solution already handles more edge cases than aws-cli, we should keep using it until we implement separate parses.

Reply:
right, but this solution only looks for one key value pair right?

Reply:
Nope, we get list of `key=value` tags. We parse each one. They are already separated by command and handled by click.